jax/_src/numpy/lax_numpy.py

```diff
@@ -3479,6 +3479,25 @@ def polyadd(a1, a2):
     return a2.at[-a1.shape[0]:].add(a1)
 
 
+@_wraps(np.polyint)
+def polyint(p, m=1, k=None):
+  m = core.concrete_or_error(operator.index, m, "'m' argument of jnp.polyint")
+  p = asarray(p)
+  if m < 0:
+    raise ValueError("Order of integral must be positive (see polyder)")
+  if k is None:
+    k = zeros(m, float)
+  k = atleast_1d(k)
+  if len(k) == 1 and m > 1:
+    k = k[0] * ones(m, float)
```

Comment:
It is slightly more efficient and avoids type promotion issues to use `k = full_like(k[0], shape=(m,))`

Reply:
This throws a TypeError. repeat(k[0], m) works, but may be slower.

Reply:
Sorry, I meant `full((m,), k[0])`